### PR TITLE
Change the logic to correctly support Hash parameters.

### PR DESCRIPTION
**Pull request by andreionut**
When a user sets a disabled job's parameters as a Hash we try to extract
the the hash from the job's arguments if the first element a Hash.

### CODE REVIEW

File: lib/resque/plugins/disable_job/job.rb [hunk 1]
```diff
@@ -93,29 +92,23 @@ def self.remove_specific_rule(rule)
...
+        def self.args_match(job_args, set_args)
+          return true if job_args == set_args
+          # We check each parameter in the job_args with the job_args to be blocked
+          # if it's nil, then we match, if it's specified, we check for equality (65 == 65)
+          should_block = if set_args.is_a?(Hash)
+            # If the set arguments is a hash we try to extract the job arguments as a hash
+            job_args = job_args.size == 1 && job_args.first.is_a?(Hash) ? job_args.first : job_args
+            job_args.map { |k, v| set_args[k].nil? || v == set_args[k] }
+          else
+            job_args.map.with_index { |a, i| set_args[i].nil? || a == set_args[i] }
```
Comment (gerardc) [inline lib/resque/plugins/disable_job/job.rb]: These comments are a smell IMO, can you extract an object to encapsulate all the checking that's going on here?

**Comment by andreionut** [comment]
@gerardc refactored the PR to move the matching logic in the `Rule` object. Can you have a second look?

File: spec/resque/plugins/disable_job/stats_spec.rb [hunk 2]
```diff
@@ -43,7 +43,7 @@ module Resque::Plugins::DisableJob
         stats.first.must_be_kind_of Rule
         stats.map(&:count).must_equal [0, 0, 0]
         # This should increment the rule counter
-        Job.disabled?(matched_job, { a: 4 })
+        p Job.disabled?(matched_job, [{ a: 4 }])
```
Comment (gerardc) [inline spec/resque/plugins/disable_job/stats_spec.rb]: you can remove the `p`

File: spec/resque/plugins/disable_job/rule_spec.rb [hunk 3]
```diff
@@ -42,5 +42,41 @@ module Resque::Plugins::DisableJob
         rule.all_rules_key.must_equal rule.main_set + ':' + name
       end
     end
+
+    describe '#match?' do
+      [
+        # job args,           set args,               should match?
+        [[], [], true],
```
Comment (gerardc) [inline spec/resque/plugins/disable_job/rule_spec.rb]: maybe worth fixing the formatting?

**Comment by gerardc** [comment]
👍 with comments

File: spec/resque/plugins/disable_job/rule_spec.rb (hunk 3 above)
Comment (andreionut) [reply spec/resque/plugins/disable_job/rule_spec.rb]: :shakes-fist: Rubocop!!!

**Pull request merged**